dehelpers/filters/cdid.py

```diff
@@ -0,0 +1,19 @@
+# ---
+# jupyter:
+#   jupytext:
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.10.2
+#   kernelspec:
+#     display_name: Python 3
+#     language: python
+#     name: python3
+# ---
+
+def cdid_filter(xycell):
+    if type(xycell.value)==str and xycell.value.isalnum() == True and xycell.value.isupper() == True and len(xycell.value) == 4:
```

Comment:
I'd maybe add something like:

```
if not isinstance(xycell.value, str):
   return False
   
if xycell.value.isalnum() == True and xycell.value.isupper() == True and len(xycell.value) == 4:
    etc ....
```
as it is, I _think_ (not 100% here) that if someone manages to pass in a xycell.value that doesn't have `isalnum()` , `isupper()` or `len()` methods you'll get an error.

not sure how you'd do that mind, but if we don't guard against it sooner or later someone will figure it out 😄